frame/composable-traits/src/staking/mod.rs

```diff
@@ -0,0 +1,158 @@
+use crate::staking::{
+	lock::{Lock, LockConfig},
+	rewards::RewardConfig,
+};
+use codec::{Decode, Encode};
+use composable_support::collections::vec::bounded::BiBoundedVec;
+
+use core::fmt::Debug;
+use frame_support::dispatch::DispatchResult;
+use scale_info::TypeInfo;
+use sp_runtime::{DispatchError, Perbill, Permill};
+
+pub mod lock;
+pub mod math;
+pub mod nft;
+pub mod rewards;
+
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub struct StakeConfig<DurationPresets, RewardsRate> {
+	pub lock: Option<LockConfig<DurationPresets>>,
+	pub reward: Option<RewardConfig<RewardsRate>>,
+}
+
+/// staking typed fNFT, usually can be mapped to raw fNFT storage type
+#[derive(Debug, PartialEq, Eq, Copy, Clone, Encode, Decode, TypeInfo)]
+pub struct Stake<Balance, Rewards> {
+	/// The original stake this NFT was minted for or updated NFT with increased stake amount.
+	pub real_stake: Balance,
+	/// List of reward asset/pending rewards.
+	pub rewards: Rewards,
+	pub lock: Option<Lock>,
+	/// The reward multiplier.
+	pub reward_multiplier: Perbill,
+}
+
+/// implemented by instances which know their share of something bigger
+pub trait Shares {
+	type Balance;
+	fn shares(&self) -> Self::Balance;
+}
+
+/// is unaware if concrete positions
+pub trait ProtocolStaking {
+	type AccountId;
+	type AssetId;
+	type Balance;
+	type PoolId;
+
+	/// Adds reward to common pool share. 
+	/// Does not actually transfers real assets.
+	fn accumulate_reward(
```

Comment:
Why is this needed in the trait?

Reply:
i am not sure then what was your comment about transfer. for me it is need as per orml-rewards + reward rate + accounts design.

Reply:
hook will call both methods + mint (default configuration to mint). so it can be done differently. will retain api with 2 methods for now. can merge later into one.

Reply:
I don't agree so prefer if we remove. As explained accumulation without minting prevents system from keeping track of total issuance needed, so I don't see a reason to "just" accumulate.